File: src/Server/base/receta_diaCena.js
const express = require('express');
const multer = require('multer');
const fs = require('fs');
const path = require('path');
const db = require('./connection');
const router = express.Router();

// Actualizar receta de cena
router.put("/cena/:id", (req, res) => {
    const { id } = req.params;
    const { fecha, idRecetaCena, idUsuarioModif } = req.body;
  
    if (!fecha || !idRecetaCena || !idUsuarioModif) {
      return res.status(400).json({ message: "Faltan datos requeridos" });
    }
  
    const query = `
      UPDATE recetas_dia
      SET Fecha = ?, Id_Receta_Cena = ?, Id_Usuario_Modif = ?, Fecha_Modif = NOW()
      WHERE Id_Recetas_Dia = ?
    `;
  
    db.query(query, [fecha, idRecetaCena, idUsuarioModif, id], (err, result) => {
      if (err) {
        console.error("Error al actualizar cena:", err);
        return res.status(500).json({ message: "Error interno del servidor" });
      }
  
      res.json({ message: "Cena actualizada con éxito" });
    });
  });
  
// Eliminar (borrar lógico) receta de desayuno
router.delete("/cena/:id", (req, res) => {
    const { id } = req.params;
    const { idUsuarioBaja } = req.body;
  
    if (!idUsuarioBaja) {
      return res.status(400).json({ message: "Faltan datos requeridos" });
    }
  
    const query = `
      UPDATE recetas_dia
      SET Id_Receta_Cena = NULL, Id_Usuario_Baja = ?, Fecha_Baja = NOW()
      WHERE Id_Recetas_Dia = ?
    `;
  
    db.query(query, [idUsuarioBaja, id], (err, result) => {
      if (err) {
        console.error("Error al eliminar cena:", err);
        return res.status(500).json({ message: "Error interno del servidor" });
      }
  
      res.json({ message: "Cena eliminada" });
    });
  });
  module.exports = router;

File: src/serverP/base/receta_diaCena.py
from fastapi import APIRouter, HTTPException, Depends
from pydantic import BaseModel
import mysql.connector
from .connection import get_db_connection

# Crear el router
recetas_dia_cena_bp = APIRouter()

# Modelos de validación con Pydantic
class IngredienteSchema(BaseModel):
    id_receta: int
    Id_Usuario_Alta: int

class EliminarSchema(BaseModel):
    Id_Usuario_Modif: int

# Actualizar receta de cena
@recetas_dia_cena_bp.put("/{Id_Recetas_Dia}")
def actualizar_cena(Id_Recetas_Dia: int, data: IngredienteSchema):
    query = """
        UPDATE recetas_dia
        SET Id_Receta_Cena = %s, Id_Usuario_Modif = %s, Fecha_Modif = NOW()
        WHERE Id_Recetas_Dia = %s
    """

    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, (data.id_receta, data.Id_Usuario_Alta, Id_Recetas_Dia))
        conn.commit()
        return {"message": "Cena actualizada con éxito"}
    except mysql.connector.Error as err:
        raise HTTPException(status_code=500, detail={"message": "Error interno del servidor", "error": str(err)})
    finally:
        cursor.close()
        conn.close()

# Eliminar (borrado lógico) receta de cena
@recetas_dia_cena_bp.put("/borrar/{Id_Recetas_Dia}")
def eliminar_cena(Id_Recetas_Dia: int, data: EliminarSchema):
    query = """
        UPDATE recetas_dia
        SET Id_Receta_Cena = NULL, Id_Usuario_Modif = %s, Fecha_Modif = NOW()
        WHERE Id_Recetas_Dia = %s
    """

    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, (data.Id_Usuario_Modif, Id_Recetas_Dia))
        conn.commit()
        return {"message": "Cena eliminada"}
    except mysql.connector.Error as err:
        raise HTTPException(status_code=500, detail={"message": "Error interno del servidor", "error": str(err)})
    finally:
        cursor.close()
        conn.close()
